### PR TITLE
docs: complete functional test scenario for sympy.solve 

### PR DESCRIPTION
**Pull request by GRIERIOR**
## Description
This PR contains the complete functional test scenario of the `sympy.solve` function following the scenario outline in `docs/functional_test.md`.
It is a foundation prior to the implementation of the runnable test.

## Checklist (DoD of the scenario from #6)
- [x] is located in `./docs/functional_test.md`
- [x] describes i) purpose of this test ii) input data iii) expected outputs
- [x] cover both regular and edge cases

## Related Issues
Refs #6

## Additional notes
The default checklist is omitted as there is no code to build or test yet.

### CODE REVIEW

File: docs/functional_test.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## Functional Test Scenario
 
+**File:** `tests/functional/test_solve_functional.py`
```
Comment (GRIERIOR) [inline docs/functional_test.md]: This file doesn't exist yet. It's weird.

File: docs/functional_test.md [hunk 2]
```diff
@@ -17,4 +19,31 @@ The goal is to catch regressions such as:
 - introducing extraneous (invalid) solutions  
 - changes in output structure that break typical usage (e.g. different formats or missing values)  
 
-Since `solve` is one of the most commonly used entry points, failures here directly translate to the perception that the library is unreliable, even if other parts remain functional.
\ No newline at end of file
+Since `solve` is one of the most commonly used entry points, failures here directly translate to the perception that the library is unreliable, even if other parts remain functional.
+
```
Comment (GRIERIOR) [inline docs/functional_test.md:21]: Unnecessary blank line.

File: docs/functional_test.md [hunk 3]
```diff
@@ -17,4 +19,31 @@ The goal is to catch regressions such as:
...
+| x**2 = -1| solve(x**2 + 1, x) | [-I, I] |
+| [x + y = 2; x = y] | solve([x + y - 2, x - y], [x, y]) | {x: 1, y: 1} |
+| [2x + y = 1; x = y] | solve([2*x + y - 1, x - y], [x, y]) | {x: 1/3, y: 1/3} |
+| x - x = 1| solve(x - x - 1) | [] |
+| 0 = 0 | solve(0) | [] |
+| sqrt(x) = -1 | solve(sqrt(x) + 1, x) | [] |
+
+### Pass criterion:
+
+- All solutions satisfy original equations when substituted (exactly or via simplification to 0)
```
Comment (GRIERIOR) [inline docs/functional_test.md]: What does it mean "exactly or via simplification to 0" - needs elaboration

File: docs/functional_test.md [hunk 4]
```diff
@@ -17,4 +19,31 @@ The goal is to catch regressions such as:
...
+|--------|----------|----------------|
+| x = -2 | solve(x + 2, x) | [-2] |
+| 3x = 9 | solve(3*x - 9, x) | [3] |
+| x**2 = 4| solve(x**2 - 4, x) | [-2, 2] |
+| x**2 = -1| solve(x**2 + 1, x) | [-I, I] |
+| [x + y = 2; x = y] | solve([x + y - 2, x - y], [x, y]) | {x: 1, y: 1} |
+| [2x + y = 1; x = y] | solve([2*x + y - 1, x - y], [x, y]) | {x: 1/3, y: 1/3} |
+| x - x = 1| solve(x - x - 1) | [] |
+| 0 = 0 | solve(0) | [] |
+| sqrt(x) = -1 | solve(sqrt(x) + 1, x) | [] |
```
Comment (GRIERIOR) [inline docs/functional_test.md]: Lack of symbolic expected results, i.e. only numerical are presents despite its crucial to test the symbolic solver.

**Comment by GRIERIOR** [comment]
For symbolic test cases it's bad idea to keep the expected result. Better to stick with simplification to 0 after substitution the solution to the original equation. It's also not quite possible then to be sure that sympy returns all the solutions - better to test it in the numerical cases as there the expected result is always unambigous.

File: docs/functional_test.md [hunk 5]
```diff
@@ -17,4 +17,61 @@ The goal is to catch regressions such as:
...
+    (R1 + R2)*I1 - R2*I2 - V1,
+    -R2*I1 + (R2 + R3 + R4)*I2 - R4*I3,
+    -R4*I2 + (R4 + R5)*I3 + V2
+], [I1, I2, I3])
+```
+
+### Pass criterion:
+
+General:
+- Output structure remains consistent with documented behavior
```
Comment (GRIERIOR) [inline docs/functional_test.md]: Missing dot.

File: docs/functional_test.md [hunk 6]
```diff
@@ -17,4 +17,61 @@ The goal is to catch regressions such as:
...
+### Pass criterion:
+
+General:
+- Output structure remains consistent with documented behavior 
+
+For numeric test cases:
+- The returned solutions match the expected result up to mathematical equivalence.
+
+For symbolic test cases:
+- All solutions satisfy original equations when substituted (using `expr.subs()` from sympy)
```
Comment (GRIERIOR) [inline docs/functional_test.md]: Missing dot.
